### DISCUSSION TITLE
Firmware update

### DISCUSSION THREAD

**Discussion by jmgiaever** [zwave-js/zwave-js-ui#747] [Q&A]
So I've been [struggling with a device ](https://github.com/zwave-js/node-zwave-js/issues/1678) and just received a file (.otz) from the vendor, and wanted to try the feature of zjs2m. However, the dialog that pops up after «begin fw update» made me unsure what to plot in as a value. It asks me to choose target, as a numeric value, but I don't understand what to fill in.

> The firmware target (i.e. chip) to upgrade. 0 updates the Z-Wave chip, >=1 updates others if they exist

I've tried to look through the manual, but it doesn't mention OTA at all, other then it supports it. 

The text «updates others if they exists» confuses me.

Should I just continue with the value 0? What happens if thats the wrong value?

**Comment by jmgiaever** [comment]
Tried with target = 0, but it ends with 

```
CNTRLR [Node 003] Firmware update timed out
INFO ZWAVE: Controller status: Node 3 firmware update FINISHED: Status -1, Time: undefined
[...]
CNTRLR [Node 003] Error while sending firmware fragment: Cannot set property 'progres
s' of undefined
[...]
CNTRLR « [Node 003] Received Firmware Update Get, but no firmware update is in progress
. Forcing the node to abort...
```

Am I doing something wrong?

**Reply by AlCalzone** [reply]
If you're not told otherwise, just use target 0, that's the main Z-Wave chip.

> CNTRLR [Node 003] Firmware update timed out

Means that the device did not request a firmware fragment in the 30 seconds after starting the update. Try moving it closer to the controller.

**Reply by jmgiaever** [reply]
Hi, thanks! I found out that if I started moving around and all my motion detectors started sending data, then it would time out. So after a few trial and errors, I ended up sitting in a dead-spot and waiting for hours to all the buggy HeatIT ZDim devices that we have been talking about earlier was updated. Took about 30 minutes each - and they're now reporting correctly! :)

**Reply by ziptbm** [reply]
Is it possible to extend the timeout time for an OTA firmware update?  maybe something greater than 30 seconds?

I'm finding a certain brand of Z-Wave 700 series switches are continually failing to update via OTA.  It takes multiple attempts, and some are still not working.  Moving closer doesn't seem to help either.   It seems like if the timeout were set longer, these might actually take.  Is that possible somehow?

**Reply by AlCalzone** [reply]
> Is it possible to extend the timeout time for an OTA firmware update? maybe something greater than 30 seconds?

Is the device in question actually requesting the firmware update to start after the 30 seconds? Do you see the corresponding command in the log?

**Reply by ziptbm** [reply]
Yes, but the system thinks it’s dead, and then after some time it’s requesting the next file so the system tells it to abort.

**Reply by AlCalzone** [reply]
Please open an issue (bug report) in https://github.com/zwave-js/node-zwave-js and provide the requested info. This sounds like something else is going on and an increased timeout isn't going to fix that.

**Reply by ziptbm** [reply]
I can do that.   I'm running ZwaveJS2MQTT via Home Assistant, but I don't believe this has anything to do with Home Assistant.   It's now isolated to ZEN77 Zooz series 700 switches and firmware updates.  The other ZEN72 and ZEN71s I have also took multiple attempts, but they eventually took the updates (much further from the controller than these 2 ZEN77s).   I've been trying periodically for almost a week, but it keeps failing.

Do I need to open something with Home Assistant first?  I don't believe this is HA related per the above.

**Reply by AlCalzone** [reply]
> Do I need to open something with Home Assistant first? I don't believe this is HA related per the above.

No

**Reply by jhein05** [reply]
I had problems with OTA fw updates too, and this has helped a bit:
1. I ping the node first and if the reply takes more than a second, it is highly likely that the update will fail
2. In order to fix it, I heal the node and ping again to see if it is faster afterwards.
3. Restart and wait an hour before trying again.
It might be hocus pocus, but I feel those things helped.

**Reply by AlCalzone** [reply]
> I ping the node first and if the reply takes more than a second

That sounds really slow.

**Reply by jhein05** [reply]
> > I ping the node first and if the reply takes more than a second
> 
> That sounds really slow.

I agree :)
